realm/kotlin-extensions/src/androidTest/kotlin/io/realm/CoroutineTests.kt

```diff
@@ -0,0 +1,147 @@
+package io.realm
+
+import io.realm.entities.SimpleClass
+import io.realm.kotlin.executeTransactionAwait
+import io.realm.kotlin.toFlow
+import io.realm.rule.BlockingLooperThread
+import io.realm.rule.TestRealmConfigurationFactory
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.util.concurrent.CountDownLatch
+import kotlin.test.*
+
+class CoroutineTests {
+
+    @Suppress("MemberVisibilityCanPrivate")
+    @Rule
+    @JvmField
+    val configFactory = TestRealmConfigurationFactory()
+
+    private lateinit var testDispatcher: TestCoroutineDispatcher
+    private lateinit var testScope: TestCoroutineScope
+    private lateinit var realm: Realm
+
+    private val looperThread = BlockingLooperThread()
+
+    @Before
+    fun setUp() {
+        realm = Realm.getInstance(configFactory.createConfiguration())
+
+        with(Realm.getDefaultInstance()) {
+            executeTransaction { it.deleteAll() }
+            close()
+        }
+
+        testDispatcher = TestCoroutineDispatcher()
+        testScope = TestCoroutineScope(testDispatcher)
+    }
+
+    @After
+    fun tearDown() {
+        realm.executeTransaction { it.deleteAll() }
+        realm.close()
+    }
+
+    @Test
+    fun toFlow_emittedOnCollect() {
+        Realm.getDefaultInstance().use { defaultRealm ->
+            defaultRealm.executeTransactionAsync {
+                it.createObject(SimpleClass::class.java).name = "Foo"
+                it.createObject(SimpleClass::class.java).name = "Bar"
+            }
+        }
+
+        val countDownLatch = CountDownLatch(1)
+
+        // TODO check this out for better testing: https://proandroiddev.com/from-rxjava-to-kotlin-flow-testing-42f1641d8433
+        val scope = CoroutineScope(Dispatchers.Main)
+        scope.launch {
+            Realm.getDefaultInstance()
+                    .where(SimpleClass::class.java)
+                    .findAll()
+                    .toFlow()
+                    .onEach { flowResults ->
+                        assertTrue(flowResults.isFrozen)
+                        if (flowResults.size == 2) {
+                            scope.cancel()
+                        }
+                    }.onCompletion {
+                        countDownLatch.countDown()
+                    }.launchIn(scope)
+        }
+        countDownLatch.await()
```

Comment:
use `TestHelper.awaitOrThrow(countDownLatch)` to avoid blocking CI for 1.5 hours if something in this test doesn't work.